fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/service/LoanReadPlatformServiceImpl.java

```diff
@@ -288,6 +290,19 @@ public LoanScheduleData retrieveRepaymentSchedule(final Long loanId,
 
     @Override
     public Page<LoanAccountData> retrieveAll(final SearchParameters searchParameters) {
+        if (searchParameters != null && searchParameters.getStatusValues() != null && searchParameters.getStatusValues().size() != 0) {
+            final List<String> statusValues = searchParameters.getStatusValues();
+            for (int i = 0; i < statusValues.size(); i++) {
+                if (LoanStatus.fromString(statusValues.get(i)) == LoanStatus.INVALID) {
+                    final List<ApiParameterError> dataValidationErrors = new ArrayList<>();
+                    final String defaultUserMessage = "The Status value '" + statusValues.get(i) + "' is not supported.";
+                    final ApiParameterError error = ApiParameterError.parameterError("validation.msg.loan.status.value.is.not.supported",
+                            defaultUserMessage, "status", statusValues.get(i));
+                    dataValidationErrors.add(error);
+                    throw new PlatformApiDataValidationException(dataValidationErrors);
+                }
+            }
+        }
```

Comment:
@vorburger  here, we are checking for all strings that were given to us and make sure none of them returns an Invalid status if it does we return a PlatformApiDataValidationException, does this answer your question? (I might have missed something?)